utils/index.d.ts

```diff
@@ -1943,3 +1943,95 @@ export declare namespace randomUtils {
     export function getRandomHexString(length?: number): string;
     export function getRandomInteger(minimumInclusive: number, maximumExclusive: number): number;
 }
+
+/**
+ * Base element for a tree view (v2)
+ */
+export declare interface TreeElementBase extends ResourceModelBase {
+    getChildren?(): ProviderResult<TreeElementBase[]>;
+    getTreeItem(): TreeItem | Thenable<TreeItem>;
+}
+
+export type TreeElementWithId = TreeElementBase & { id: string };
```

Comment:
This is the first time `TreeElementBase` is being introduced into utils. Client extensions using the v2 API are not forced to use this model. However, we will be writing all of our own utils based on this model. Thus, our own extensions don't _have to_, but _should_ use this model when they consume the v2 api.

model model model